### PR TITLE
feat: reactions on Messages(sqservices -1734)

### PR DESCRIPTION
**Pull request by arjita-mitra**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - As a Wire user I would like to have a wide range of reactions (emojis) at my disposal, so that I can use them to express myself in messages and reactions to messages.

- The **PR Description**
  - create message actions floating menu
  - integrate context menu, reply, emoji picker
  - manage action menu states focus/hover/selected
  - handle outside click
  - make everything keyboard accessible
  - unit testing
----
##### References
1. https://wearezeta.atlassian.net/browse/SQSERVICES-1734

### CODE REVIEW

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx [hunk 1]
```diff
@@ -185,21 +174,41 @@ const ContentMessageComponent: React.FC<ContentMessageProps> = ({
...
+  const [isActionMenuVisible, setActionMenuVisibility] = useState(true);
+  const isMenuOpen = useMessageActionsState(state => state.isMenuOpen);
+  const isReactionFeatureEnabled = Config.getConfig().FEATURE.ENABLE_REACTION;
+
+  useEffect(() => {
+    if (isMessageFocused || msgFocusState) {
+      setActionMenuVisibility(true);
+    } else {
+      setActionMenuVisibility(false);
+    }
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx:144]: I think you can use shorter version :)
setActionMenuVisibility(isMessageFocused || msgFocusState)

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  handleKeyDown,
+  resetActionMenuStates,
+  wrapperRef,
+}) => {
+  const isThumbUpAction = currentMsgActionName === MessageActionsId.THUMBSUP;
+  const isLikeAction = currentMsgActionName === MessageActionsId.HEART;
+  const [showEmojis, setShowEmojis] = useState(false);
+  const {handleMenuOpen} = useMessageActionsState();
+  const [clientX, setPOSX] = useState(0);
+  const [clientY, setPOSY] = useState(0);
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions.tsx]: Move this 0 above component, e.g. const INITIAL_CLIENT_X_POS = 0;

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        aria-label={t('accessibility.messageActionsMenuThumbsUp')}
+        data-uie-name={MessageActionsId.THUMBSUP}
+        aria-pressed={isThumbUpAction}
+        type="button"
+        tabIndex={messageFocusedTabIndex}
+        onClick={handleMsgActionClick}
+        onKeyDown={handleMsgActionKeyDown}
+      >
+        <span aria-hidden={true}>👍</span>
+      </button>
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions.tsx]: Please add smth like that as props also: data-stringify-emoji=":thumbsup:"

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx (hunk 1 above)
Comment (arjita-mitra) [reply src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx:144]: with if, else its more readable so keeping it this way

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions.tsx (hunk 3 above)
Comment (arjita-mitra) [reply src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions.tsx]: why do we need to add this prop?

**Review by atomrc — APPROVED** [review]

File: yarn.lock
Comment (atomrc) [inline yarn.lock:1]: Please delete that file, should be there

**Review by atomrc — CHANGES_REQUESTED** [review]

File: yalc.lock [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (atomrc) [inline yalc.lock]: This file shoundn't be commited, please remove

**Pull request merged**